Add privacy-and-robustness-in-lightning review to chaincode-labs/chaincode-podcast (#482)
bitcointranscripts/bitcointranscripts: chaincode-labs/chaincode-podcast/privacy-and-robustness-in-lightning.md

---
title: 'Privacy and robustness in Lightning'
transcript_by: 'markon1-a via review.btctranscripts.com'
media: 'https://podcasters.spotify.com/pod/show/chaincode/episodes/Rusty-Russell-and-privacy-and-robustness-in-Lightning---episode-34-e27t6go'
date: '2023-08-09'
tags:
  - 'lightning'
  - 'eltoo'
  - 'privacy-enhancements'
speakers:
  - 'Rusty Russell'
categories:
  - 'podcast'
episode: 34
---
Rusty Russell: 00:00:00

Simplicity is a virtue in itself, right?
So you can try to increase fairness.
If you reduce simplicity, you end up actually often stepping backwards.
The thing that really appeals about element symmetry is it's simple.
And every scheme to fix things by introducing more fairness and try to reintroduce penalties and everything else ends up destroying the simplicity.

> Speaker 1: 00:00:27

Author: I was not able to identify the two speakers.

> Speaker 1: 00:00:27

Reviewer: The host and cohost for chaincode podcast are **Adam Jonas** and **Mark Erhardt**

> Speaker 1: 00:00:27

Author: Okay thanks. Corrected the speakers.

But also, I think like the recent fee spike showed us that there are definitely places we can improve.
Robustness from the protocol level means we want stuff where we become more immune to fee issues.
I think that's something definitely, we need longer term.
The assumption is fees are going to go up, they're going to get more volatile, we're going to have these spikes, we need to handle that really well.
And I certainly hope that we, we have a number of tricks we can do.
Some of which is like, well, we want some soft forks, right?

## LN symmetry

Rusty Russell: 00:04:39

If we can get LN symmetry, which used to be called L2, but LN symmetry is a much better name.
Sorry to question who named L2.
He blames me.
I blame him.
For example, sidesteps a whole heap of, it reduces robustness by introducing simplicity.
It does make it a lot simpler to maintain your Lightning channel.
There's no longer toxic waste that you have to worry about, old states and stuff like that.
Watchtower has become now almost trivial.
So that adds some simplicity.

Rusty Russell: 00:05:05

Can we take a quick side route on that?

Speaker 2:

Sure.

> Rusty Russell:

Reviewer: Missing timestamps. Try to set the correct timestamps

Adam Jonas: 00:16:10

When we were talking about  Eltoo, you're talking about changes and you're sort of casually mentioning the soft fork piece of it.
I mean, there's not going well, there's Greg Sanders will be there and he's obviously thinking about this, but you don't really have the people at the table that are actually doing like would be pulling the soft fork together that really understand the lightning case, I guess.
Maybe AJ, maybe Instagibbs.
The list gets very short after that.
So, like, how are you able to communicate those needs down to the lower layer?

Rusty Russell: 00:16:47

Well, I think the any prep out cases have been pretty strong for a long time.
The question's always been, do we want other soft fork as well?

## CTV

Rusty Russell: 00:16:54

And people are like, oh, if we had obstacle the `CHECKSIG` from stack, we can also do it.
Sure.
I can probably build a car out of matchsticks and snot, but I don't want to.
And don't make me do it.

Adam Jonas: 00:17:09

Could you do that?

Rusty Russell: 00:17:12

A theoretical engineer probably could do that.
I could not do that.
I'm not crafty at all.
But, okay.
That is probably an aside.
We don't need to go down.

> But, you know, `CHECKSIG` from stack is like, it's cool that you can even make that work.

Reviewer 2: This actually refers to the opcode `OP_CHECKSIGFROMSTACK`.
Make sure to also change it in the rest of the transcript.

But it is a terrible basis for just about anything.
And it's more like a game we could play as far as what's the most obscure way we could do some kind of introspection.
And the answer would be, `CHECKSIG` from stack.
And the proposals that are in that cluster, and there are a few, are pretty clean, pretty well understood.
AJ would definitely be the person unfortunately he's not here in New York he's kind of, done a fair bit of travel recently he was like no I'm not going.

> So I'm hoping that while we're in New York he is in his mountain lair somewhere coming up with a scheme to activate any prep out.

Reviewer 2: ```suggestion
So I'm hoping that while we're in New York he is in his mountain lair somewhere coming up with a scheme to activate `ANYPREVOUT`.
```

That's if you're listening, AJ, I hope you're going to make that prophecy come true.
I think, we understand that we want this and we've wanted it for a fair period of time and nothing else has come up that's gone, oh no, we want this instead.
Any prep out by itself is a fairly simple piece.

Adam Jonas: 00:18:21

Not a precluding piece to other things that we can do in the future.

Rusty Russell: 00:18:24

Absolutely.

Adam Jonas: 00:18:25

This is the two-way door, one-way door kind of thing.

Rusty Russell: 00:18:28

> In this case, we will want any prep out even when we have full covenants and everything else.

Reviewer 2: Same here, this refers to `ANYPREVOUT`, full name is `SIGHASH_ANYPREVOUT` but because speakers refer to it as `ANYPREVOUT`, it's okay to write just that.

So I'm pretty confident to say this is something that we should enable.
Right?

> Now that said the detail the fine details you know hand wave hand wave I know try not to spend too much time coming over details but I'm pretty happy that, somebody will come up with a proposal.

Reviewer 2: I appreciate removing the "you know" filer, but the extra comma "," is not required
```suggestion
Now that said the detail the fine details you know hand wave hand wave I know try not to spend too much time coming over details but I'm pretty happy that somebody will come up with a proposal.
```

Adam Jonas: 00:16:10

When we were talking about  Eltoo, you're talking about changes and you're sort of casually mentioning the soft fork piece of it.
I mean, there's not going well, there's Greg Sanders will be there and he's obviously thinking about this, but you don't really have the people at the table that are actually doing like would be pulling the soft fork together that really understand the lightning case, I guess.
Maybe AJ, maybe Instagibbs.
The list gets very short after that.
So, like, how are you able to communicate those needs down to the lower layer?

Rusty Russell: 00:16:47

Well, I think the `ANYPREVOUT` cases have been pretty strong for a long time.
The question's always been, do we want other soft fork as well?

## CTV

Rusty Russell: 00:16:54

And people are like, oh, if we had `OP_CTV` and `OP_CHECKSIGFROMSTACK`, we can also do it.
Sure.
I can probably build a car out of matchsticks and snot, but I don't want to.
And don't make me do it.

Adam Jonas: 00:17:09

Could you do that?

Rusty Russell: 00:17:12

A theoretical engineer probably could do that.
I could not do that.
I'm not crafty at all.
But, okay.
That is probably an aside.
We don't need to go down.

> But,`CHECKSIG` from stack is like, it's cool that you can even make that work.

Reviewer: The `OP_CHECKSIGFROMSTACK` correction by @kouloumos still applies here

Reviewer: Make the change and commit

Author: Already did ealier.

Reviewer 2: you didn't make this change, please see this commit to understand what we were asking for: https://github.com/bitcointranscripts/bitcointranscripts/pull/482/commits/9eda4c9e97f7ff98505827591be656f81a2833ac

But it is a terrible basis for just about anything.
And it's more like a game we could play as far as what's the most obscure way we could do some kind of introspection.

> And the answer would be, `CHECKSIG` from stack.

Reviewer: Here also

Author: Noted

Reviewer 2: you didn't make this change, please see this commit to understand what we were asking for: https://github.com/bitcointranscripts/bitcointranscripts/pull/482/commits/9eda4c9e97f7ff98505827591be656f81a2833ac

Should all of us fail?
We've had this discussion, right?
She's been like, cool, that's okay.
You can go do that.
At least for a while before you have to get a real job again.
So I feel fairly confident to go, we're going to move forward.
And maybe not as fast as we are at the moment.
But, you know, there's some future there.

Adam Jonas: 00:23:24

I guess my other bit of that was Because they're commercially associated, maybe not driven, commercially associated.
That means that features get pushed to the front of the line.
Because I've worked for companies, I know how it works.
So when you're talking about, ring fencing things and making sure that we're, thinking long term.
These are different than just like building a product that's, in JavaScript and, can wow an investor.
There is a philosophical side to what's happening here.
And there is a collaboration side of no implementation can do it by themselves.

Rusty Russell: 00:24:07

Yes.

Adam Jonas: 00:24:07

And so those dynamics just don't seem to add up to a long term thinking and long term, moving, moving the ball forward in these kinds of dimensions.

> Rusty Russell: 00:24:21

Reviewer 2: Here you deleted the "you know" filler word, but you could also delete the duplicate phrase to make it more readable
```suggestion
And so those dynamics just don't seem to add up to a long term thinking and long term moving the ball forward in these kinds of dimensions.
```
